### DISCUSSION TITLE
Error: bad sample width

### DISCUSSION THREAD

**Discussion by efoxxfiles** [JonathanFly/bark#66] [General]
Hi

Today bark infinity stop to work
Any idea?

tqdm._instances.clear() 
full_generation_segments, audio_arr_segments, final_filename_will_be = api.generate_audio_long(text_prompt=Texto, history_prompt=history_prompt)

Audio(np.concatenate(audio_arr_segments), rate=generation.SAMPLE_RATE) 

Error                                     Traceback (most recent call last)

[<ipython-input-6-e569ffa86ba4>](https://localhost:8080/#) in <cell line: 8>()
      6 
      7 tqdm._instances.clear()
----> 8 full_generation_segments, audio_arr_segments, final_filename_will_be = api.generate_audio_long(text_prompt=Texto, history_prompt=history_prompt)
      9 
     10 Audio(np.concatenate(audio_arr_segments), rate=generation.SAMPLE_RATE)

5 frames

[/usr/lib/python3.10/wave.py](https://localhost:8080/#) in setsampwidth(self, sampwidth)
    351             raise Error('cannot change parameters after starting to write')
    352         if sampwidth < 1 or sampwidth > 4:
--> 353             raise Error('bad sample width')
    354         self._sampwidth = sampwidth
    355 

Error: bad sample width

**Comment by DiamondGlassDrill** [comment]
I get the same error.

Traceback (most recent call last):
  File "K:\Voice\bark\bark_perform.py", line 136, in <module>
    main(namespace_args)
  File "K:\Voice\bark\bark_perform.py", line 121, in main
    api.generate_audio_long(**args_dict)
  File "K:\Voice\bark\bark_infinity\api.py", line 1099, in generate_audio_long
    write_one_segment(audio_arr = np.concatenate(audio_arr_segments), full_generation = full_generation_segments[0], **kwargs)
  File "K:\Voice\bark\bark_infinity\api.py", line 360, in write_one_segment
    write_seg_wav(filepath, audio_arr, **kwargs)
  File "K:\Voice\bark\bark_infinity\api.py", line 409, in write_seg_wav
    write_audiofile(filepath, audio_arr, **kwargs)
  File "K:\Voice\bark\bark_infinity\api.py", line 438, in write_audiofile
    audio.export(output_filepath, format="mp4", codec="aac")
  File "C:\mambaforge\envs\clonevoice\lib\site-packages\pydub\audio_segment.py", line 890, in export
    wave_data.setsampwidth(self.sample_width)
  File "C:\mambaforge\envs\clonevoice\lib\wave.py", line 353, in setsampwidth
    raise Error('bad sample width')
wave.Error: bad sample width
Exception ignored in: <function Wave_write.__del__ at 0x000001F0B8798A60>

All generated .mp4 or .mp3 files are empty, where the .npz are being generated and contain data

**Comment by JonathanFly** [comment]
I think I have a fix. Make sure you aren't using MP4 format for now. I'll be online tomorrow Sunday evening will patch.

**Comment by JonathanFly** [comment]
This should be fixed. Let me know if works. 

```git pull``` to update

As far I can tell you only got the error if you tried to add silence between segments. Or if you used the CLI, where I accidentally left it defaulting to adding 0.25 seconds of silence.

**Comment by DiamondGlassDrill** [comment]
File is still empty. What I do is typing a text message and select a build in speaker, than I click generate it, it produces the .npz and an .mp3 but the .mp3 is empty.

**Reply by JonathanFly** [reply]
> File is still empty. What I do is typing a text message and select a build in speaker, than I click generate it, it produces the .npz and an .mp3 but the .mp3 is empty.

Can you try setting the output format to 'wav' just to check that works at least?

Also just to double check you got the update, you typed `git pull` from the command line, same place you start python bark_webui.py?
